ziti/cmd/ziti/cmd/lets_encrypt/accounts_storage.go

```diff
@@ -0,0 +1,254 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package lets_encrypt
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-acme/lego/v4/certcrypto"
+	"github.com/go-acme/lego/v4/lego"
+	"github.com/go-acme/lego/v4/log"
+	"github.com/go-acme/lego/v4/registration"
+)
+
+const (
+	baseAccountsRootFolderName = "accounts"
+	baseKeysFolderName         = "keys"
+	accountFileName            = "account.json"
+)
+
+// AccountsStorage A storage for account data.
+//
+// rootPath:
+//
+//     ./.path/accounts/
+//          │      └── root accounts directory
+//          └── "path" option
+//
+// rootUserPath:
+//
+//     ./.path/accounts/localhost_14000/hubert@hubert.com/
+//          │      │             │             └── userID ("email" option)
+//          │      │             └── CA server ("server" option)
+//          │      └── root accounts directory
+//          └── "path" option
+//
+// keysPath:
+//
+//     ./.path/accounts/localhost_14000/hubert@hubert.com/keys/
+//          │      │             │             │           └── root keys directory
+//          │      │             │             └── userID ("email" option)
+//          │      │             └── CA server ("server" option)
+//          │      └── root accounts directory
+//          └── "path" option
+//
+// accountFilePath:
+//
+//     ./.path/accounts/localhost_14000/hubert@hubert.com/account.json
+//          │      │             │             │             └── account file
+//          │      │             │             └── userID ("email" option)
+//          │      │             └── CA server ("server" option)
+//          │      └── root accounts directory
+//          └── "path" option
```

Comment:
Is this something we came up with, or was this something that is a standard, or a defacto standard, or something else?